src/components/UploadingJobsDisplay/UploadingJobsDisplay.js

```diff
@@ -127,19 +165,58 @@ class UploadingJobsDisplay extends Component {
   };
 
   onFileUploadSuccess = ({ file }) => {
-    this.updateFileState(file, { uploadStatus: UPLOADED });
+    this.updateFileState(file, {
+      status: fileStatuses.UPLOADED,
+      uploadDate: new Date(), // TODO replace with field from backend (MODDATAIMP-54)
+    });
   };
 
   onFileUploadFail = ({ file }) => {
-    this.updateFileState(file, { uploadStatus: FAILED });
+    this.updateFileState(file, { status: fileStatuses.FAILED });
+  };
+
+  handleUndoDeleteFile = key => {
+    const file = this.state.files[key];
+
+    clearTimeout(this.deleteFileTimeouts[key]);
+
+    this.updateFileState(file, { status: fileStatuses.UPLOADED });
+  };
+
+  handleDeleteFile = key => {
+    const { timeoutBeforeFileDeletion } = this.props;
+    const file = this.state.files[key];
+
+    this.deleteFileTimeouts[key] = setTimeout(() => {
+      API.deleteFile(
+        this.deleteFileUrl(file),
+        this.createDeleteFileHeaders(),
+      )
+        .then(() => this.deleteFileFromState(key))
+        .catch(error => {
+          this.updateFileState(file, { status: fileStatuses.UPLOADED });
+          console.error(error); // eslint-disable-line no-console
+        });
+    }, timeoutBeforeFileDeletion);
+
+    this.updateFileState(file, { status: fileStatuses.DELETING });
+  };
+
+  deleteFileFromState = key => {
+    const { files } = this.state;
+    const updatedFiles = Object.assign({}, files);
```

Comment:
Consider rewriting with spread operator instead
```suggestion
    const updatedFiles = { ...files };
```